contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -777,19 +801,141 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractEr
+                        // If we are here, our pre-propose module has
+                        // errored while receiving a proposal
+                        // hook. Rest in peace pre-propose module.
+                        CREATION_POLICY.save(deps.storage, &ProposalCreationPolicy::Anyone {})?;
+                        addr
+                    }
+                }
+            } else {
+                // The index we're getting a response to is one of our
+                // regular proposal hooks.
+                PROPOSAL_HOOKS.remove_hook_by_index(deps.storage, idx)?
+            };
+
+            Ok(Response::new().add_attribute("removed_proposal_hook", format!("{addr}:{idx}")))
         }
         TaggedReplyId::FailedVoteHook(idx) => {
             let addr = VOTE_HOOKS.remove_hook_by_index(deps.storage, idx)?;
             Ok(Response::new().add_attribute("removed vote hook", format!("{addr}:{idx}")))
         }
+        TaggedReplyId::PreProposeModuleInstantiation => {
+            let res = parse_reply_instantiate_data(msg)?;
+            let module = deps.api.addr_validate(&res.contract_address)?;
+            CREATION_POLICY.save(
+                deps.storage,
+                &ProposalCreationPolicy::Module { addr: module },
+            )?;
+
+            Ok(Response::new().add_attribute("update_pre_propose_module", res.contract_address))
+        }
     }
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
     // Set contract to version to latest
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-    Ok(Response::default())
+    match msg {
+        MigrateMsg::FromV1 {
+            close_proposal_on_execution_failure,
+            pre_propose_info,
+        } => {
+            // Update the stored config to have the new
+            // `close_proposal_on_execution_falure` field.
+            let current_config = v1_state::CONFIG.load(deps.storage)?;
+            CONFIG.save(
+                deps.storage,
+                &Config {
+                    voting_strategy: current_config.voting_strategy,
+                    max_voting_period: current_config.max_voting_period,
+                    min_voting_period: current_config.min_voting_period,
+                    only_members_execute: current_config.only_members_execute,
+                    allow_revoting: current_config.allow_revoting,
+                    dao: current_config.dao,
+                    // Loads of text, but we're only updating these fields.
+                    close_proposal_on_execution_failure,
+                },
+            )?;
+
+            let (initial_policy, pre_propose_messages) =
+                pre_propose_info.into_initial_policy_and_messages(env.contract.address.clone())?;
```

Comment:
most recent updates make it so that `into_initial_policy_and_messages` expects the cw-dao-core address here. reason being that it instantiates the pre-propose module to have that address as its CosmWasm level admin.

```suggestion
                pre_propose_info.into_initial_policy_and_messages(current_config.dao)?;
```

also need to add a clone above to please the borrow checker:
```rust
dao: current_config.dao.clone(),
```